gym/envs/registration.py

```diff
@@ -516,6 +518,14 @@ def make(
     if autoreset:
         env = AutoResetWrapper(env)
 
+    if not disable_env_checker:
```

Comment:
what is the reason for having that as opt-out instead of opt-in?

user will see a double reset from his/her side and step into the env, which might be quite confusing (for instance using real robot) and/or costly.

Reply:
FWIW I initially wanted to stick to a non-invasive checker that only looks at function signatures, but the usefulness of that was pretty limited. 

The main purpose of having it opt-out is sort of educational - to let people know if the environment isn't correctly implemented, basically saying "Hey, it looks like it works alright, but here's a few missing elements, so you might want to fix it". If it were opt-in, then it would be largely useless, since you might as well run the checker manually

Reply:
btw, the PR still mentions "What I strongly do not want to do here is actually call any methods like reset or step which would be necessary for a thorough API verification. This would have unintended consequences while creating the env, and that's a bad thing. It should be 100% unintrusive."

but you went the other way at the end?

Reply:
> but you went the other way at the end?

I didn't retroactively change the original PR message, so essentially that was just my thought process at the time. Essentially I had (and still have) the same points you're stating here.

In the meantime, there were some "offline" discussions where the conclusion was to use the full env checker and make it opt-out (based on what I described earlier). Which for me is a bit "**ugh**, fine", as in I didn't veto it, but I don't love it.

Reply:
as a compromise would it be possible to let the user define a global parameter/env variable that would disable that new feature?
something like `gym.disable_auto_check()` or `os.environ["DISABLE_AUTO_CHECK"] = 1`

because with gym 0.24, I now have to patch all pieces of code to `gym.make(..., disable_env_checker=False)` to ensure old behavior (including at test time).

btw, this behavior will probably break video recording too...

Reply:
So the reason we added this as default on was ultimately that even before the API changes, virtually no environments code complied with the Gym API, which is bad for standardization, the field, and so on. The intention of this change was to make this known is a very obvious manner so that things are actually fixed.

You're correct that for environments where resetting is undesirable they'll have to add a flag, but this is a minority of the plethora of environments. As far as a global parameter/variable or breaking video recording, let me get back to you

Reply:
I like the idea of a global switch, either environment variable or a global setting in gym, I'll have to think a bit which one would be better.